src/services/mongodb/collection.js

```diff
@@ -19,47 +19,43 @@ class Collection {
    *
    * @method
    * @param {Object} doc The document to insert.
-   * @param {Object} [options] Additional options object.
    * @return {Promise<Object, Error>} a Promise for the operation.
    */
-  insertOne(doc, options = {}) {
-    return insertOp(this, doc, options);
+  insertOne(doc) {
+    return insertOneOp(this, doc);
   }
 
   /**
    * Inserts multiple documents.
    *
    * @method
    * @param {Array} docs The documents to insert.
-   * @param {Object} [options] Additional options object.
    * @return {Promise<Object, Error>} Returns a Promise for the operation.
    */
-  insertMany(docs, options = {}) {
-    return insertOp(this, docs, options);
+  insertMany(docs) {
+    return insertManyOp(this, docs);
   }
 
   /**
    * Deletes a single document.
    *
    * @method
    * @param {Object} query The query used to match a single document.
-   * @param {Object} [options] Additional options object.
    * @return {Promise<Object, Error>} Returns a Promise for the operation.
    */
-  deleteOne(query, options = {}) {
-    return deleteOp(this, query, Object.assign({}, options, { singleDoc: true }));
+  deleteOne(query) {
+    return deleteOp(this, false, query);
   }
 
   /**
    * Deletes all documents matching query.
    *
    * @method
    * @param {Object} query The query used to match the documents to delete.
-   * @param {Object} [options] Additional options object.
    * @return {Promise<Object, Error>} Returns a Promise for the operation.
    */
-  deleteMany(query, options = {}) {
-    return deleteOp(this, query, Object.assign({}, options, { singleDoc: false }));
+  deleteMany(query) {
+    return deleteOp(this, true, query);
```

Comment:
i don't know if we get much out of having `deleteOp` as a helper anymore (it's just a one-liner now).
I think you could remove it and just call `return buildResponse("deleteMany", this, buildArgs(service, { query })))` directly. Same for `deleteOne` and `insertOne`/`insertMany` too.